### PR TITLE
test(core): Add unit tests for customer-group promotion condition

### PR DESCRIPTION
**Pull request by GabrielRoc**
## Description

Adds a direct unit test suite for the `customer_group` promotion condition (`customer-group-condition.ts`), which previously had **0% function coverage**.

The condition checks whether `order.customer` belongs to a given `CustomerGroup` by calling `CustomerService.getCustomerGroups()` through a `CacheService`-backed cache and then comparing IDs with `idsAreEqual`. Three NestJS services (`CustomerService`, `CacheService`, `EventBus`) are injected into the condition's module-level variables via `init()` — the test supplies lightweight mocks, with the cache configured as a pass-through so group lookups always hit the mock service directly.

6 tests covering (equivalence partitioning):
- Order with no customer → `false`
- Customer with no groups → `false`
- Customer in other groups but not the target → `false`
- Customer in exactly the target group → `true`
- Customer in multiple groups including the target → `true`
- Correct forwarding of `customerId` and `ctx` to `getCustomerGroups()`

All tests are database-free.

## Breaking changes

None.

## Checklist

- [x] I have set a clear title
- [x] My PR is small and contains a single feature
- [x] I have added or updated test cases

Relates to #4835

---
<a href="https://app.blacksmith.sh/vendurehq/codesmith/vendure/pr/4883"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1785127906&installation_id=128408429&pr_number=4883&repository=vendurehq%2Fvendure&return_to=https%3A%2F%2Fgithub.com%2Fvendurehq%2Fvendure%2Fpull%2F4883&signature=229e1843aeb09016a4351397d3a7df488a67b9be0b78b735fa0bcf96353eb512"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by GabrielRoc** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/core/src/config/promotion/conditions/customer-group-condition.spec.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    ofType: () => of(),
+} as unknown as EventBus;
+
+const mockInjector = {
+    get: (token: unknown) => {
+        if (token === CustomerService) return mockCustomerService;
+        if (token === CacheService) return mockCacheService;
+        if (token === EventBus) return mockEventBus;
+    },
+};
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/customer-group-condition.spec.ts:54]: Valid concern. Fixed in commit `180431e1f`: replaced `token === CustomerService` / `token === CacheService` / `token === EventBus` with constructor-name matching (`(token as { name?: string })?.name === 'CustomerService'` etc.). Constructor names are stable regardless of whether the module was resolved via a `.js`-suffixed dynamic import or a bare static import, so the injector mock is now robust to ESM module-identity differences under Vitest. The now-unused static imports of those three classes were also removed.

File: packages/core/src/config/promotion/conditions/customer-group-condition.spec.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+function orderWithCustomer(customerId: ID): Order {
+    return new Order({ lines: [], customer: new Customer({ id: customerId }) });
+}
+
+const ctx = createRequestContext({ pricesIncludeTax: false });
+
+async function check(order: Order, args: ConfigArg[]) {
+    return customerGroup.check(ctx, order, args, undefined as any);
+}
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/customer-group-condition.spec.ts:72]: Dismissed. `customerGroup.check()` does not use the `promotion` argument. Passing `undefined as any` is the consistent pattern across all promotion-condition specs in this PR series.

**Review by grolmus — APPROVED** [review]
Excellent, well-structured coverage — boundary-value analysis on the thresholds, clean injector-based mocking (no DB), and assertions that the condition is invoked with the right args. Test-only, no production changes. Approving.
